### PR TITLE
Split out the modifiers into their own files

### PR DESCRIPTION
**Pull request by graywolf336**
This is the first step in starting #89 again, since this would be some conflicts.

### CODE REVIEW

File: src/definition/accessors/ILivechatMessageBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    /**
+     * Sets the emoji to use for the avatar, this overwrites the current avatar
+     * whether it be the user's or the avatar url provided.
+     *
+     * @param emoji the emoji code
+     */
+    setEmojiAvatar(emoji: string): ILivechatMessageBuilder;
+
+    /**
+     * Gets the emoji used for the avatar.
```
Comment (lolimay) [inline src/definition/accessors/ILivechatMessageBuilder.ts]:
```suggestion
     * Gets the emoji code used for the avatar.
```

File: src/definition/accessors/IRoomBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+     * @param username the user's username to add to the room
+     * @deprecated in favor of `addMemberByUsername`. This method will be removed on version 2.0.0
+     */
+    addUsername(username: string): IRoomBuilder;
+
+    /**
+     * Sets the usernames of who are joined to the room.
+     *
+     * @param usernames the list of usernames
+     * @deprecated in favor of `setMembersByUsernames`. This method will be removed on version 2.0.0
```
Comment (lolimay) [inline src/definition/accessors/IRoomBuilder.ts]: Do you mean `addMemberToBeAddedByUsername`?

https://github.com/RocketChat/Rocket.Chat.Apps-engine/blob/25790ce041d5056bf9f6d0507dde9a4dc36bd9b7/src/definition/accessors/IRoomBuilder.ts#L92-L97

File: src/definition/accessors/IRoomBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+     */
+    setUsernames(usernames: Array<string>): IRoomBuilder;
+
+    /**
+     * Gets the usernames of users in the room.
+     * @deprecated in favor of `getMembersUsernames`. This method will be removed on version 2.0.0
```
Comment (lolimay) [inline src/definition/accessors/IRoomBuilder.ts]: Do you mean `setMembersToBeAddedByUsernames`?

File: src/definition/accessors/IRoomBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    /**
+     * Gets the room's creator.
+     */
+    getCreator(): IUser;
+
+    /**
+     * Adds a user to the room, these are by username until further notice.
+     *
+     * @param username the user's username to add to the room
+     * @deprecated in favor of `addMemberByUsername`. This method will be removed on version 2.0.0
```
Comment (lolimay) [inline src/definition/accessors/IRoomBuilder.ts]: Do you mean `addMemberToBeAddedByUsername`?

https://github.com/RocketChat/Rocket.Chat.Apps-engine/blob/25790ce041d5056bf9f6d0507dde9a4dc36bd9b7/src/definition/accessors/IRoomBuilder.ts#L92-L97

Comment (lolimay) [inline src/definition/accessors/IRoomBuilder.ts]: Do you mean `addMemberToBeAddedByUsername`?

https://github.com/RocketChat/Rocket.Chat.Apps-engine/blob/25790ce041d5056bf9f6d0507dde9a4dc36bd9b7/src/definition/accessors/IRoomBuilder.ts#L92-L97

File: src/definition/accessors/IRoomBuilder.ts (hunk 2 above)
Comment (lolimay) [inline src/definition/accessors/IRoomBuilder.ts]: Do you mean `setMembersToBeAddedByUsernames`?

File: src/definition/accessors/IRoomBuilder.ts (hunk 3 above)
Comment (lolimay) [inline src/definition/accessors/IRoomBuilder.ts]: So you mean `getMembersToBeAddedUsernames`?

https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/151/files#diff-49dcbc2c88e63aeb2a1dfc6647690d1aR109

**Comment by d-gubert** [comment]
@lolimay as this PR is just the splitting of the content of one big file, I think we can leave the refactoring for later.

Thanks for the review, though!

**Pull request merged**